parteh/PRTAllometricCarbonMod.F90

```diff
@@ -0,0 +1,981 @@
+    call bdead_allom( target_agw_c, target_bgw_c, target_sapw_c, ipft, target_struct_c)
+
+
+    ! ------------------------------------------------------------------------------------
+    ! If structure is larger than target, then we need to correct some integration errors
+    ! by slightly increasing dbh to match it.
+    ! For grasses, if leaf biomass is larger than target, then we reset dbh to match
+    ! -----------------------------------------------------------------------------------
+    if( (( struct_c - target_struct_c ) > calloc_abs_error) .and. &
+          (EDPftvarcon_inst%woody(ipft) == itrue) ) then
+
+       call StructureResetOfDH( struct_c, ipft, &
+             canopy_trim, dbh, hite_out )
+
+       ! Target sapwood biomass and deriv. according to allometry and trimming [kgC, kgC/cm]
+       call bsap_allom(dbh,ipft,canopy_trim,sapw_area,target_sapw_c)
+
+       ! Target total above ground deriv. biomass in woody/fibrous tissues  [kgC, kgC/cm]
+       call bagw_allom(dbh,ipft,target_agw_c)
+       
+       ! Target total below ground deriv. biomass in woody/fibrous tissues [kgC, kgC/cm] 
+       call bbgw_allom(dbh,ipft,target_bgw_c)
+       
+       ! Target total dead (structrual) biomass and deriv. [kgC, kgC/cm]
+       call bdead_allom( target_agw_c, target_bgw_c, target_sapw_c, ipft, target_struct_c)
+
+    end if
+    
+    ! Target leaf biomass according to allometry and trimming
+    call bleaf(dbh,ipft,canopy_trim,target_leaf_c)
+
+    ! Target fine-root biomass and deriv. according to allometry and trimming [kgC, kgC/cm]
+    call bfineroot(dbh,ipft,canopy_trim,target_fnrt_c)
+
+    ! Target storage carbon [kgC,kgC/cm]
+    call bstore_allom(dbh,ipft,canopy_trim,target_store_c)
+
+    
+    leaf_c0 = leaf_c
+    fnrt_c0 = fnrt_c
+    sapw_c0 = sapw_c
+    store_c0 = store_c
+    repro_c0 = repro_c
+    struct_c0 = struct_c
+
+    ! -----------------------------------------------------------------------------------
+    ! III.  Prioritize some amount of carbon to replace leaf/root turnover
+    !         Make sure it isnt a negative payment, and either pay what is available
+    !         or forcefully pay from storage. 
+    ! -----------------------------------------------------------------------------------
+    
+    if( EDPftvarcon_inst%evergreen(ipft) ==1 ) then
+       leaf_c_demand   = max(0.0_r8, &
+             EDPftvarcon_inst%leaf_stor_priority(ipft)*this%variables(leaf_c_id)%turnover(icd))
+    else
+       leaf_c_demand   = 0.0_r8
+    end if
+    
+    fnrt_c_demand   = max(0.0_r8, &
+          EDPftvarcon_inst%leaf_stor_priority(ipft)*this%variables(fnrt_c_id)%turnover(icd))
+
+    total_c_demand = leaf_c_demand + fnrt_c_demand
+    
+    if (total_c_demand> nearzero ) then
+
+       ! If we are testing b4b, then we pay this even if we don't have the carbon
+       ! Just don't pay so much carbon that storage+carbon_balance can't pay for it
```

Comment:
So, once the B4B tests are done then this would/should be altered to be more reasonable when the demand is too high?

Reply:
yeah, I think that was what I was getting at.  There is no check there to specify what is a reasonable amount to transfer.  It's as if this part of the code implicitly trusts that the respiration costs (and therefore negative carbon gains) will be reasonable for how much storage is available.  

It will use up all of its storage carbon to pay the leaf and root turnover costs if it has to.